### PR TITLE
Delete user

### PR DESCRIPTION
**Pull request by tcitworld**
| Q             | A
| ------------- | ---
| Bug fix?      |no
| New feature?  | yes
| BC breaks?    |no
| Deprecations? |no
| Tests pass?   | yes
| Documentation |no
| Translation   | yes
| Fixed tickets | #1640
| License       | MIT

### CODE REVIEW

**Comment by nicosomb** [comment]
Need tests?

**Comment by tcitworld** [comment]
Yup, and not working properly right now.

**Comment by nicosomb** [comment]
We need to add a confirmation step too.

**Comment by j0k3r** [comment]
Rebased ✅

**Comment by nicosomb** [comment]
If I'm an admin, I don't want to see this button in my config page.

**Comment by tcitworld** [comment]
> If I'm an admin, I don't want to see this button in my config page.

Why ? There can be multiple admin accounts and an admin has not necessary shell access to do this as he/she likes.

**Comment by nicosomb** [comment]
Today, we can't specify who is an admin. Only the first user is an admin, so if we delete him, we can create an other one.

**Comment by tcitworld** [comment]
In that case, we should count the admin accounts and hide the button if this admin is the only one.

**Comment by nicosomb** [comment]
why not. The easiest solution is the best for me.

**Comment by tcitworld** [comment]
> why not. The easiest solution is the better one for me.

It's to prepare the future. :-)

File: tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php [hunk 1]
```diff
@@ -649,4 +649,89 @@ public function testDemoMode()
...
+    }
+
+    public function testDeleteAccount()
+    {
+        $this->logInAs('wallace');
+        $client = $this->getClient();
+
+        $crawler = $client->request('GET', '/config');
+
+        $deleteLink = $crawler->filter('.red')->last()->link();
```
Comment (j0k3r) [inline tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php]: Can't there be a better selector which are not going to fail when we'll add a new `red` class on an element?

File: tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php [hunk 2]
```diff
@@ -649,4 +649,89 @@ public function testDemoMode()
...
+        $em = $client->getContainer()->get('doctrine.orm.entity_manager');
+        $user = $em
+            ->getRepository('WallabagUserBundle:User')
+            ->createQueryBuilder('u')
+            ->where('u.username = :username')->setParameter('username', 'wallace')
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+
+        $this->assertTrue(false !== $user);
```
Comment (j0k3r) [inline tests/Wallabag/CoreBundle/Controller/ConfigControllerTest.php]: `assertNotNull($user)` ?

**Comment by nicosomb** [comment]
@modos189 We need popin like here https://github.com/wallabag/wallabag/issues/1640 Can you help us please?

File: src/Wallabag/UserBundle/Entity/User.php [hunk 3]
```diff
@@ -240,4 +240,14 @@ public function isTrustedComputer($token)
 
         return false;
     }
+
+    public function serialize()
+    {
+        return serialize($this->id);
```
Comment (j0k3r) [inline src/Wallabag/UserBundle/Entity/User.php]: @tcitworld why do we need to serialize the user?

**Comment by j0k3r** [comment]
Here are some preview:

File: src/Wallabag/UserBundle/Entity/User.php (hunk 3 above)
Comment (tcitworld) [inline src/Wallabag/UserBundle/Entity/User.php]: I don't remember exactly but I guess it was required for something. Please test without it.

Comment (j0k3r) [inline src/Wallabag/UserBundle/Entity/User.php]: Removed it changed nothing.

**Comment by tcitworld** [comment]
Looks good.

**Pull request merged**